sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala

```diff
@@ -295,6 +298,13 @@ class EthService(val scNodeViewHolderRef: ActorRef, val nvtimeout: FiniteDuratio
     }
   }
 
+  private def getBlockAtTag[A](nodeView: NV, tag: String)(fun: AccountBlock ⇒ A): A = {
+    nodeView.history.getBlockById(getBlockIdByTag(nodeView, tag)).asScala match {
+      case None => throw new RpcException(RpcError.fromCode(RpcCode.InvalidParams, "Invalid block tag parameter."))
+      case Some(block) => fun(block)
```

Comment:
This function style structure is useful in the other method because want to wrap the `fun` execution in a `using` to make sure we properly close the view. Here you could also just return the block.